cmd/snap/cmd_run.go

```diff
@@ -249,5 +261,15 @@ func runSnapConfine(info *snap.Info, securityTag, snapApp, command, hook string,
 	cmd = append(cmd, snapApp)
 	cmd = append(cmd, args...)
 
+	// FIXME2: for now "ForceDevMode" will prevent snap-confine to be
+	//         run on distros like fedora/suse etc. which is ok because
+	//         snap-confine is a non-op on these distros anyway.
```

Comment:
Yes, you are correct. Thinking about this, we cannot re-exec and not use snap-confine from core as well. Things get our of sync and there will be havoc. We need to distable re-exec (like we do on some distros currently) in this case.